### PR TITLE
Upload transcript page functions + modal

### PR DESCRIPTION
**Pull request by noorilu**
# Description

**BIG IDEA:** After a user successfully uploads a transcript on the upload transcript screen, have a modal pop-up on clicking the "Begin Session" button that allows users to enter their desired flow name which leads to the starting intent screen (if successful).

**ADDED FUNCTIONALITIES:**
- Made modals a generic component
- Allows users to upload transcripts: If the file isn't a JSON file/not in the correct format, prompts alert on the screen asking for a valid file. If it's a duplicate, prompts alert pop-up saying it has already been uploaded.
- On clicking "Begin Session", opens the modal for users to enter the flow name. This also stores the response id of createTranscript after uploading the transcript to use in other functions later on (like deleting the transcript). This is done by updating sessionProvider.js for transcript ID state.

- **On the modal:**
**PRESSING CONTINUE:** If no flow name has been entered, prompts alert pop-up. Otherwise, stores the flow name that user entered with the transcript file onto DB. Then, it takes us to starting intent page. 
**PRESSING CANCEL:** delete the transcript uploaded onto the DB and allow users to select different file (or keep the same one if they changed their minds).

**Fixes # (issue)**

- transcripts.js file promises correctly returns desired variables now
- Uploading the flow correctly passes in a JSON file now instead of a string
- Error with button height size changing on upload transcript page fixed
- Modified test transcript file to only go into 3 depths (4 depths makes transcript file too long to upload onto the DB since we are using a free version)

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Checking MongoDB to see if objects are added and deleted when desired
- [x] Using console.log to see if certain functions and variables are the values we want
- [x] Running it on localhost and clicking certain buttons to make sure everything shows up on the UI as intended

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by noorilu** [comment]
just waiting until i get a chance to communicate with group members --- i have never encountered a merge conflict but should i just resolve it like it tells me to above?

git pull origin integration
git checkout lucy/upload-transcript-page-functions
git merge integration
git push -u origin lucy/upload-transcript-page-functions

**Comment by noorilu** [comment]
Just commenting that the "fixing CSS styling" branch was made off of this branch so it had most of my changes written in the description of this PR. This PR will be a really quick review if anyone can just look at it for 5 min.

File: mern/client/src/Contexts/sessionProvider.js [hunk 1]
```diff
@@ -6,10 +6,18 @@ export default function SessionProvider({ children }) {
   // nav state to show or hide session id/save/delete buttons
   const [navState, setNavState] = React.useState(false);
   const [transcriptID, setTranscriptID] = React.useState();
```
Comment (jkolyakov) [inline mern/client/src/Contexts/sessionProvider.js:8]: This file context is getting quite big in terms number of states. Are all of them necessary to be in sessionProvider? Also what is the difference between navState and transcriptID

File: mern/client/src/Contexts/sessionProvider.js [hunk 2]
```diff
@@ -6,10 +6,18 @@ export default function SessionProvider({ children }) {
...
+  const [firstQuestions, setfirstQuestions] = React.useState();
```
Comment (jkolyakov) [inline mern/client/src/Contexts/sessionProvider.js:9]: Also, add a comment here saying what the state is/it's purpose. Easier for understanding.

**Review by jkolyakov — APPROVED** [review]
LGTM. Just one question about the context and a note about commenting to do before merging.

File: mern/client/src/Contexts/sessionProvider.js (hunk 1 above)
Comment (noorilu) [reply mern/client/src/Contexts/sessionProvider.js:8]: Yeah maybe I can move the transcriptID to a different context file? I felt that it was relevant to the session so I just kept it there for now. 

The difference between navState and transcriptID is that the navState is either true or false (when true, it shows the save and delete buttons on the top, when false, it hides it), and for the transcriptID, I'm using to store the string of the transcriptID the creator uploads so we can pass it into functions such as deleting the transcript from the DB.

File: mern/client/src/Contexts/sessionProvider.js (hunk 2 above)
Comment (noorilu) [reply mern/client/src/Contexts/sessionProvider.js:9]: Good idea!

File: mern/client/src/Contexts/sessionProvider.js (hunk 1 above)
Comment (jkolyakov) [reply mern/client/src/Contexts/sessionProvider.js:8]: But can't we just check if transcriptID is defined to see if the session has been entered?

Comment (jkolyakov) [reply mern/client/src/Contexts/sessionProvider.js:8]: Or is it better for the single responsibility principle to split each of those things into a different state?

Comment (noorilu) [reply mern/client/src/Contexts/sessionProvider.js:8]: Hmmmmmmm now I'm thinking, the only reason you would want the navState to not show is when you haven't started a session meaning no transcript has been uploaded.......... I think you're right, we can check if transcriptID is defined for the navState. Unless recover session does something different? Does recovering the session set the transcriptID?

**Pull request merged**